week-1-ts-ecosystem/w01e04-light-me-up/src/modules/lightBulb.module.ts

```diff
@@ -0,0 +1,25 @@
+import { PowerSource } from "./powerSourceModule";
+const timer = () => new Promise((res) => setTimeout(res, 1000));
+
+export class LightBulb {
+  protected readonly powerConsumption = 20;
+  powerSource: PowerSource;
+
+  constructor(powerSource: PowerSource) {
+    this.powerSource = powerSource;
+  }
+  async light(time: number = 1) {
+    if (time) {
```

Comment:
To `if(time)` mi się nie podoba, ponieważ "oplata" cały kod.
Nie lepiej odwrócić warunek ?

Zauważ również że nie tylko `time` jako `> 1` zwróci tutaj `true` ale również ` < 1 ` !!! tylko `0` zwróci false.
Co prawda pętla nie będzie się obracać ale skoro chcesz się @kubaGalinski zabezpieczyć. To może tak:

```javascript
if (time <= 0) {
return;
}
```
I temat załatwiony, a potem lecimy z pętlą i logiką.

Reply:
tutaj bez poprawki @kubaGalinski?